### ISSUE TITLE
Internal double quotes in values fail to parse

### ISSUE THREAD

**Issue by jtdaugherty** [aisamanra/config-ini#40]
Hey @aisamanra, I hope you're doing well! I got this report a while back on Matterhorn and I'd love to know what you make of it. https://github.com/matterhorn-chat/matterhorn/issues/761

**Comment by aisamanra** [comment]
I am, and also hope you're doing well too!

For what it's worth, it doesn't seem like it's a problem with the core INI-parsing machinery, at least here in the source-to-raw-format step: taking the exact line in that PR and adding it to a test seems to produce a successful parse: c.f. the added test in #41 (which succeeds in every test configuration I've got.) I haven't pushed it, but I've created a bidir example that uses that exact bad example as a value and it also works.

That said, it looks like Matterhorn has [a lot of stuff built on top of `config-ini` that I haven't seen before](https://github.com/matterhorn-chat/matterhorn/blob/master/src/Matterhorn/Config/Schema.hs). Is it possible that it parses the underlying raw INI correctly and then runs into an issue there?
